src/Repository/LinkBlockRepository.php

```diff
@@ -0,0 +1,269 @@
+    {
+        $maxPosition = $this->getHookMaxPosition($idHook);
+
+        $qb = $this->connection->createQueryBuilder();
+        $qb
+            ->insert($this->dbPrefix.'link_block')
+            ->values([
+                'id_hook' => ':idHook',
+                'position' => ':position',
+                'content' => ':content',
+            ])
+            ->setParameters([
+                'idHook' => $idHook,
+                'position' => $maxPosition + 1,
+                'content' => json_encode([
+                    'cms' => empty($cms) ? [false] : $cms,
+                    'static' => empty($static) ? [false] : $static,
+                    'product' => empty($product) ? [false] : $product,
+                ]),
+            ]);
+        ;
+        $this->executeQueryBuilder($qb, 'Link block error: ');
+        $linkBlockId = $this->connection->lastInsertId();
+
+        $this->updateLanguages($linkBlockId, $blockName, $custom);
+
+        return $linkBlockId;
+    }
+
+    /**
+     * @param int   $linkBlockId
+     * @param array $blockName
+     * @param int   $idHook
+     * @param array $cms
+     * @param array $static
+     * @param array $product
+     * @param array $custom
+     * @return string
+     * @throws PrestaShopDatabaseException
+     */
+    public function update($linkBlockId, array $blockName, $idHook, array $cms, array $static, array $product, array $custom)
+    {
+        $qb = $this->connection->createQueryBuilder();
+        $qb
+            ->update($this->dbPrefix.'link_block', 'lb')
+            ->andWhere('lb.id_link_block = :linkBlockId')
+            ->set('id_hook', ':idHook')
+            ->set('content', ':content')
+            ->setParameters([
+                'linkBlockId' => $linkBlockId,
+                'idHook' => $idHook,
+                'content' => json_encode([
+                    'cms' => empty($cms) ? [false] : $cms,
+                    'static' => empty($static) ? [false] : $static,
+                    'product' => empty($product) ? [false] : $product,
+                ]),
+            ])
+        ;
+        $this->executeQueryBuilder($qb, 'Link block error: ');
+
+        $this->updateLanguages($linkBlockId, $blockName, $custom);
+
+        return $linkBlockId;
+    }
+
+    /**
+     * @param int $idLinkBlock
+     * @throws PrestaShopDatabaseException
+     */
+    public function delete($idLinkBlock)
+    {
+        $tableNames = [
+            'link_block_shop',
+            'link_block_lang',
+            'link_block',
+        ];
+
+        foreach ($tableNames as $tableName) {
+            $qb = $this->connection->createQueryBuilder();
+            $qb
+                ->delete($this->dbPrefix.$tableName)
+                ->andWhere('id_link_block = :idLinkBlock')
+                ->setParameter('idLinkBlock', $idLinkBlock)
+            ;
+            $this->executeQueryBuilder($qb, 'Delete error: ');
```

Comment:
Not a fan of using space here. should you add this in 
`throw new PrestaShopDatabaseException($errorPrefix . ' ' . var_export($statement->errorInfo(), true));` ?